tools/rpkg/R/backend-dbplyr__duckdb_connection.R

```diff
@@ -0,0 +1,392 @@
+        }
+      },
+      log10 = sql_prefix("LOG10", 1),
+      log2 = sql_prefix("LOG2", 1),
+
+      # See https://github.com/duckdb/duckdb/issues/530 about NaN, infinites and NULL in DuckDB
+      # The following is how R functions for detecting those should behave:
+      # Function 	    Inf 	–Inf 	NaN 	NA
+      # is.finite() 	FALSE FALSE FALSE FALSE
+      # is.infinite() TRUE 	TRUE 	FALSE FALSE
+      # is.nan() 	    FALSE FALSE TRUE 	FALSE
+      # is.na() 	    FALSE FALSE TRUE 	TRUE
+      is.na = function(a) build_sql("(", a, " IS NULL OR PRINTF('%f',", a, ") = 'nan')"),
+      is.nan = function(a) build_sql("(", a, " IS NOT NULL AND PRINTF('%f',", a, ") = 'nan')"),
+      is.infinite = function(a) build_sql("(", a, " IS NOT NULL AND REGEXP_MATCHES(PRINTF('%f',", a, "),'inf'))"),
+      is.finite = function(a) build_sql("(NOT (", a, " IS NULL OR REGEXP_MATCHES(PRINTF('%f',", a, "),'inf|nan')))"),
+      grepl = duckdb_grepl,
+
+      # Return index where the first match starts,-1 if no match
+      regexpr = function(p, x) {
+        build_sql("(CASE WHEN REGEXP_MATCHES(", x," ,", p, ") THEN (LENGTH(LIST_EXTRACT(STRING_SPLIT_REGEX(", x, ", ", p,"),0))+1) ELSE -1 END)")
+      },
+      round = function(x, digits) sql_expr(ROUND(!!x, CAST(ROUND((!!digits), 0L) %AS% INTEGER))),
+      as.Date = sql_cast("DATE"),
+      as.POSIXct = sql_cast("TIMESTAMP"),
+
+      # lubridate functions
+
+      month = function(x, label = FALSE, abbr = TRUE) {
+        if (!label) {
+          sql_expr(EXTRACT(MONTH %FROM% !!x))
+        } else {
+          if (abbr) {
+            sql_expr(STRFTIME(!!x, "%b"))
+          } else {
+            sql_expr(STRFTIME(!!x, "%B"))
+          }
+        }
+      },
+      quarter = function(x, type = "quarter", fiscal_start = 1, with_year = identical(type, "year.quarter")) {
+        if (fiscal_start != 1) {
+          stop("`fiscal_start` is not yet supported in DuckDB translation. Must be 1.", call. = FALSE)
+        }
+        if (is.logical(type)) {
+          type <- if (type) {
+            "year.quarter"
+          } else {
+            "quarter"
+          }
+        }
+        if (with_year) {
+          type <- "year.quarter"
+        }
+        switch(type,
+          quarter = {
+            sql_expr(EXTRACT(QUARTER %FROM% !!x))
+          },
+          year.quarter = {
+            sql_expr((EXTRACT(YEAR %FROM% !!x) || "." || EXTRACT(QUARTER %FROM% !!x)))
+          },
+          date_first = {
+            sql_expr((CAST(DATE_TRUNC("QUARTER", !!x) %AS% DATE)))
+          },
+          date_last = {
+            build_sql("(CAST((DATE_TRUNC('QUARTER',", x, ") + INTERVAL '1 QUARTER' - INTERVAL '1 DAY') AS DATE))")
+          },
+          stop(paste("Unsupported type", type), call. = FALSE)
+        )
+      },
+      qday = function(x) {
+        build_sql("DATE_DIFF('DAYS',DATE_TRUNC('QUARTER',CAST((", x, ") AS DATE)),(CAST((", x, ") AS DATE) + INTERVAL '1 DAY'))")
```

Comment:
The rest of the codebase seems to add spaces after commas.

```suggestion
        build_sql("DATE_DIFF('DAYS', DATE_TRUNC('QUARTER',CAST((", x, ") AS DATE)), (CAST((", x, ") AS DATE) + INTERVAL '1 DAY'))")
```